fix: add a note about experimental windows support (#19644)
docker/docs: content/build/buildkit/_index.md

> **Warning**
>

> > BuildKit only supports building Linux containers. Now there is experimental
> > Windows support that is tracked in

Reviewer: ```suggestion
> BuildKit only fully supports building Linux containers. Experimental Windows
> container support is tracked in
```
